README.md

```diff
@@ -6,18 +11,47 @@ This project is developed and maintained by the [HAL team][team].
 
 ## [Documentation](https://docs.rs/nb)
 
+The ultimate goal of this crate is *code reuse*. With this crate you can
+write *core* I/O APIs that can then be adapted to operate in either blocking
+or non-blocking manner. Furthermore those APIs are not tied to a particular
+asynchronous model and can be adapted to work with the `futures` model or
+with the `async` / `await` model.
```

Comment:
side note: Time has shown this is a bit of a lie, perhaps we could update it. For example you can't adapt `nb` to `async/await` because it has no wakers :upside_down_face: . Also the terminology below about `futures`/`await` is super outdated.

Reply:
True. Could you make a wording update PR?